ui/store/actions.ts

```diff
@@ -732,29 +732,23 @@ export function signPersonalMsg(
 
 export function decryptMsgInline(
   decryptedMsgData: TemporaryMessageDataType['msgParams'],
-): ThunkAction<
-  Promise<TemporaryMessageDataType>,
-  MetaMaskReduxState,
-  unknown,
-  AnyAction
-> {
+): ThunkAction<Promise<string | null>, MetaMaskReduxState, unknown, AnyAction> {
   log.debug('action - decryptMsgInline');
   return async (dispatch: MetaMaskReduxDispatch) => {
     log.debug(`actions calling background.decryptMessageInline`);
 
-    let newState;
+    let rawMessage = null;
     try {
-      newState = await submitRequestToBackground<
-        MetaMaskReduxState['metamask']
-      >('decryptMessageInline', [decryptedMsgData]);
+      rawMessage = await submitRequestToBackground<string>(
+        'decryptMessageInline',
+        [decryptedMsgData],
+      );
     } catch (error) {
       logErrorWithMessage(error);
       dispatch(displayWarning(error));
       throw error;
     }
-
-    dispatch(updateMetamaskState(newState));
```

Comment:
We don't really need to update MM state here, because UI takes that `rawMessage` and set it on local state.